### PR TITLE
DBZ-1442 Porting insert source fields feature to ExtractNewDocumentState SMT

### PR DESCRIPTION
**Pull request by ChingTsai**
https://issues.jboss.org/projects/DBZ/issues/DBZ-1442

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Thanks, I'll look into this asap. No worries about the test failure, it's unrelated in a different connector.

**Comment by gunnarmorling** [comment]
Getting back to this now; sorry for the delay, we were at a face-to-face meeting last week.

File: debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java [hunk 1]
```diff
@@ -92,6 +93,41 @@ public void closeSmt() {
         transformation.close();
     }
 
+    @Test
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java]: Could you add the `@FixFor` annotation?

File: debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java [hunk 2]
```diff
@@ -92,6 +93,41 @@ public void closeSmt() {
...
+        // when
+        SourceRecord transformed = transformation.apply(record);
+        Struct value = (Struct) transformed.value();
+
+        // assert source fields' value
+        assertThat(value.get("__h")).isEqualTo(12345678L);
+        assertThat(value.get("__ts_ms")).isEqualTo(1000000L);
+        assertThat(value.get("__ord")).isEqualTo(1);
+        assertThat(value.get("__db")).isEqualTo("dbA");
+        assertThat(value.get("__rs")).isEqualTo("rs0");
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java]: Could you also add a test for a delete event?

**Comment by ChingTsai** [comment]
@gunnarmorling  No problem, I will work on it.

**Comment by gunnarmorling** [comment]
Thanks a lot! If you see the chance for doing it today, we could include this in tomorrow's 0.10.0.CR1 release. But no worries if it's not doable, it'll mean not too much of deferral otherwise.

**Comment by renatomefi** [comment]
I might be a bit too late to the party, but wouldn't that be better served as headers?

And if not, have you considered using the native `insertField` smt instead of writing a custom behavior? https://docs.confluent.io/current/connect/transforms/insertfield.html#insertfield

Thanks a lot for this contribution, I'll probably use to as well! :)

**Comment by gunnarmorling** [comment]
Hey @renatomefi,

>  wouldn't that be better served as headers?

The idea is to have these fields as a part of the actual record in the sink. I don't think reading such values from headers is commonly supported by popular connectors? Note this implementation is adding the same feature that already exists for the relational counterpart to this SMT.

> have you considered using the native insertField

Would this work, as I reckon it doesn't work with nested fields (essentially, you'd have to propagate e.g. "source.txId" to "after.txId", which I don't think is supported).

**Comment by gunnarmorling** [comment]
Rebased and applied. Also added one more commit for describing the new option in the docs. Thanks a lot, @ChingTsai!

**Pull request closed without merging**